packet.go

```diff
@@ -314,5 +396,107 @@ func (p *Packet) decodeIp6(recur int) {
 		p.decodeIcmp()
 	case IP_INIP:
 		p.decodeIp6(recur + 1)
+	case IP_GRE:
+		p.decodeGre()
+	}
+}
+func (p *Packet) decodeVxLAN() bool {
+	// VxLAN decode is a heuristic and not an exact algorithm.
+
```

Comment:
As I mentioned above, we should have two functions, one to validate and the other to decode.

Reply:
added validation for ether, ip and ip6.